src/hlsBinaries.ts

```diff
@@ -499,6 +531,44 @@ export function addPathToProcessPath(extraPath: string): string {
     return PATH.join(pathSep);
 }
 
+// complements getLatestHLSfromMetadata, by checking possibly locally compiled
+// HLS in ghcup
+// If 'targetGhc' is omitted, picks the latest 'haskell-language-server-wrapper',
+// otherwise ensures the specified GHC is supported.
+async function getLatestHLSfromGHCup(
+  context: ExtensionContext,
+  storagePath: string,
+  logger: Logger,
+  targetGhc?: string
```

Comment:
`targetGhc` is undefined for the first pass, where we just want the hls wrapper and don't care about hls exe.